src/sidebar/services/test/session-test.js

```diff
@@ -31,47 +23,59 @@ describe('sidebar/services/session', function () {
       }),
     };
     fakeAuth = Object.assign(new EventEmitter(), {
-      login: sandbox.stub().returns(Promise.resolve()),
+      login: sinon.stub().returns(Promise.resolve()),
       logout: sinon.stub().resolves(),
     });
     fakeSentry = {
-      setUserInfo: sandbox.spy(),
+      setUserInfo: sinon.spy(),
     };
     fakeApi = {
       profile: {
-        read: sandbox.stub().resolves(),
-        update: sandbox.stub().resolves({}),
+        read: sinon.stub().resolves(),
+        update: sinon.stub().resolves({}),
       },
     };
     fakeServiceConfig = sinon.stub().returns(null);
     fakeSettings = {
       serviceUrl: 'https://test.hypothes.is/root/',
     };
-    fakeToastMessenger = { error: sandbox.spy() };
+    fakeToastMessenger = { error: sinon.spy() };
+
+    const retryPromiseOperation = async callback => {
+      // eslint-disable-next-line no-constant-condition
+      while (true) {
+        try {
+          return await callback();
+        } catch (err) {
+          // Try again
+        }
+      }
+    };
```

Comment:
This mock matches how `retryPromiseOperation` delays are removed in `APIRoutesService` tests.